packages/cdktf/lib/tokens/token.ts

```diff
@@ -82,6 +82,62 @@ export class Token {
     );
   }
 
+  /**
+   * Return a reversible map representation of this token
+   */
+  public static asMap(
+    value: any,
+    mapValue: any,
+    options: EncodingOptions = {}
+  ): { [key: string]: any } {
+    // since the return value is basically an object, just encode always
+    return TokenMap.instance().registerMap(
+      Token.asAny(value),
+      mapValue,
+      options.displayHint
+    );
+  }
+
+  /**
+   * Return a reversible map representation of this token
+   */
+  public static asStringMap(
+    value: any,
+    options: EncodingOptions = {}
+  ): { [key: string]: string } {
+    return this.asMap(value, "", options);
```

Comment:
If I understand this correctly `""` is the value used in the map, e..g `{ "&[Token.1]": ""}`. I wonder if we can make this string more telling that it's part of a token value and not a real string, so that if someone accidentally accesses e.g. all values of a token and sees an empty string they have some more clues to what is going on. Long-term we should check this value when resolving outside of a map context, similar to the warning we show when accessing a value from within a list, but that is probably out of scope here

Reply:
Sounds good. I initially had it as the inner token part, but changed when needing to make type specific versions.
Easy to use any string here, so I'll make a constant value that is more token like. I can also put in a dummy value for number maps.
If the check seems simple enough, I'll just add it; otherwise, I'll put in a follow up PR.

Reply:
Sounds great, thank you 👍